src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2204,6 +2443,381 @@ def construct_command(self, task_exec, task_args, task_numcores,
+
+            logger.info('Stoping DFS.')
+            os.system(lm_info['hadoop_home'] + '/sbin/stop-dfs.sh')
+
+            logger.info("Deleting HADOOP files from temp")
+            os.system('rm -rf /tmp/hadoop*')
+            os.system('rm -rf /tmp/Jetty*')
+            os.system('rm -rf /tmp/hsperf*')
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        # Single Node configuration
+        # TODO : Multinode config
+        self._log.info('Getting YARN app')
+        os.system('wget http://eceweb1.rutgers.edu/~ip176/yarn/Pilot-YARN-0.1-jar-with-dependencies.jar')
+        self._log.info(self._cfg['lrms_info']['lm_info'])
+        self.launch_command = self._cfg['lrms_info']['lm_info']['launch_command']
+        self._log.info('YARN was called')
+        
+
+    # --------------------------------------------------------------------------
+    #
+    def construct_command(self, task_exec, task_args, task_numcores,
+                          launch_script_hop, opaque_slots, (cu_descr,work_dir)):
+
+        # Construct the args_string which is the arguments given as input to the
+        # shell script. Needs to be a string
+        self._log.debug("Constructing YARN command")
+
+        self._log.debug('Opaque Slots {0}'.format(opaque_slots))
+
+        if 'lm_info' not in opaque_slots:
+            raise RuntimeError('No lm_info to launch via %s: %s' \
+                    % (self.name, opaque_slots))
+
+        if not opaque_slots['lm_info']:
+            raise RuntimeError('lm_info missing for %s: %s' \
+                               % (self.name, opaque_slots))
+
+        if 'service_url' not in opaque_slots['lm_info']:
+            raise RuntimeError('service_url not in lm_info for %s: %s' \
+                    % (self.name, opaque_slots))
+
+        if 'rm_url' not in opaque_slots['lm_info']:
+            raise RuntimeError('rm_url not in lm_info for %s: %s' \
+                    % (self.name, opaque_slots))
+
+
+        if 'nodename' not in opaque_slots['lm_info']:
+            raise RuntimeError('nodename not in lm_info for %s: %s' \
+                    % (self.name, opaque_slots))
+
+        service_url = opaque_slots['lm_info']['service_url']
+        rm_url      = opaque_slots['lm_info']['rm_url']
+        client_node = opaque_slots['lm_info']['nodename']
+
+
+        #-----------------------------------------------------------------------
+        # Create YARN script
+        # This funcion creates the necessary script for the execution of the
+        # CU's workload in a YARN application. The function is responsible
+        # to set all the necessary variables, stage in, stage out and create
+        # the execution command that will run in the distributed shell that
+        # the YARN application provides. There reason for staging out is
+        # because after the YARN application has finished everything will be
+        # deleted.
```

Comment:
I may be missing it, but is the CU `environment` set anywhere in the resulting script?